### PR TITLE
feature/RefactorClasses for Easier Code Coverage Testing

### PR DESCRIPTION
**Pull request by BeansJeesh**

### CODE REVIEW

File: Example/SampleMerchantApp/SampleMerchantApp.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -7,7 +7,7 @@
 	objects = {
 
 /* Begin PBXBuildFile section */
```
Comment (cesar-oyarzun-m) [inline Example/SampleMerchantApp/SampleMerchantApp.xcodeproj/project.pbxproj]: Revert this file, don't push any pbxproj file unless you're adding new files

File: MCSCommerceWebTests/MCSWebViewControllerManagerTests.m [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+#import "MockViewController.h"
+
+@interface MCSWebViewControllerManagerTests : XCTestCase
+
+@end
+
+@implementation MCSWebViewControllerManagerTests
+
+-(void)testWebViewController {
+    MockViewController *mockVc = [[MockViewController alloc] init];
```
Comment (cesar-oyarzun-m) [inline MCSCommerceWebTests/MCSWebViewControllerManagerTests.m]: Add prefix MCS to any new class

File: MCSCommerceWebTests/MCSWebViewControllerManagerTests.m [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+//    window.rootViewController = mockVc;
```
Comment (cesar-oyarzun-m) [inline MCSCommerceWebTests/MCSWebViewControllerManagerTests.m]: you can remove this

File: Podfile [hunk 4]
```diff
@@ -3,4 +3,12 @@ use_frameworks!
 target 'MCSCommerceWeb' do
   pod 'SVGKit', :git => 'https://github.com/SVGKit/SVGKit.git', :branch => '3.x'
 
-end
\ No newline at end of file
+  pod "CocoaLumberjack"
```
Comment (cesar-oyarzun-m) [inline Podfile]: Remove CocoaLumberjack, we're already using DXPLogger as dependency

File: README.md [hunk 5]
```diff
@@ -11,6 +11,8 @@
 
 `MCSCommerceWeb` is a lightweight framework used to integrate Merchants with [**EMV Secure Remote Commerce**](https://www.emvco.com/emv-technologies/src/) and Mastercard's web-based SRC-Initiator with backward compatibility for existing Masterpass integrations. `MCSCommerceWeb` facilitates the initiation of the checkout experience and returns the transaction result to the Merchant after completion.
 
+***Note: currently, this framework is only recommended for existing U.S. Masterpass merchants. Merchants using version 2.8 must follow the steps in the migration section. Merchants using version older than 2.8 should follow all steps wihtout migration from this integration guide.***
```
Comment (cesar-oyarzun-m) [inline README.md]: I think you didn't branch out of develop, we remove this note in our readme file

**Review by cesar-oyarzun-m — COMMENTED** [review]
I think we need to test using zip file to make sure everything works, please upload to a new branch the zipfile so we can test.

File: MCSCommerceWebTests/MCSWebViewControllerManagerTests.m (hunk 2 above)
Comment (BeansJeesh) [reply MCSCommerceWebTests/MCSWebViewControllerManagerTests.m]: got it

File: MCSCommerceWebTests/MCSWebViewControllerManagerTests.m (hunk 3 above)
Comment (BeansJeesh) [reply MCSCommerceWebTests/MCSWebViewControllerManagerTests.m]: done

File: README.md (hunk 5 above)
Comment (BeansJeesh) [reply README.md]: cloned another time from develop to check the note, but it was there. I can remove the note though

File: Podfile (hunk 4 above)
Comment (BeansJeesh) [reply Podfile]: removed

File: Example/FancyShop/Pods/Target Support Files/Pods-FancyShop/Pods-FancyShop-frameworks.sh [hunk 6]
```diff
@@ -46,8 +46,8 @@ install_framework()
   fi
 
   # Use filter instead of exclude so missing patterns don't throw errors.
```
Comment (cesar-oyarzun-m) [inline Example/FancyShop/Pods/Target Support Files/Pods-FancyShop/Pods-FancyShop-frameworks.sh]: You can revert this file, actually we should stop tracking anything inside Pods folder

File: Example/FancyShop/FancyShop.xcodeproj/project.pbxproj [hunk 7]
```diff
@@ -21,8 +21,6 @@
 		19A0EDB322971653006B7AFA /* LaunchScreen.storyboard in Resources */ = {isa = PBXBuildFile; fileRef = 19A0EC6B22971651006B7AFA /* LaunchScreen.storyboard */; };
 		19A0EDB422971653006B7AFA /* AppDelegate.swift in Sources */ = {isa = PBXBuildFile; fileRef = 19A0EC6D22971651006B7AFA /* AppDelegate.swift */; };
 		19A0EDB622971653006B7AFA /* Constants.swift in Sources */ = {isa = PBXBuildFile; fileRef = 19A0EC7122971651006B7AFA /* Constants.swift */; };
-		19A0EDBC22971653006B7AFA /* Prod.xcconfig in Resources */ = {isa = PBXBuildFile; fileRef = 19A0EC7822971651006B7AFA /* Prod.xcconfig */; };
-		19A0EDBD22971653006B7AFA /* Sandbox.xcconfig in Resources */ = {isa = PBXBuildFile; fileRef = 19A0EC7922971651006B7AFA /* Sandbox.xcconfig */; };
```
Comment (cesar-oyarzun-m) [inline Example/FancyShop/FancyShop.xcodeproj/project.pbxproj:25]: Revert this file

File: MCSCommerceWebTests/MCSWebViewControllerManagerTests.m [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+//
+
+#import <XCTest/XCTest.h>
+#import "MCSWebViewControllerManager.h"
+#import "MCSCheckoutUrlBuilder.h"
+#import "MCSConfigurationManager.h"
+#import "MCSConfiguration.h"
+#import "MCSCommerceweb.h"
+#import "MCSCheckoutRouter.h"
+#import "MockViewController.h"
```
Comment (cesar-oyarzun-m) [inline MCSCommerceWebTests/MCSWebViewControllerManagerTests.m]: Add prefix MCS

Comment (BeansJeesh) [reply MCSCommerceWebTests/MCSWebViewControllerManagerTests.m]: added MCS prefix

File: Example/FancyShop/FancyShop/Modules/OrderSummary/Application Logic/Interactor/OrderSummaryInteractor.swift [hunk 9]
```diff
@@ -136,7 +136,8 @@ class OrderSummaryInteractor:BaseInteractor, OrderSummaryInteractorInputProtocol
     }
     
     func initializeSdk() {
-        self.APIDataManager?.initializeSdk()
+//        self.APIDataManager?.initializeSdk(with: (self.presenter as? OrderSummaryPresenter)?.view as? UIViewController)
```
Comment (cesar-oyarzun-m) [inline Example/FancyShop/FancyShop/Modules/OrderSummary/Application Logic/Interactor/OrderSummaryInteractor.swift]: Add comment this is an example

File: Example/FancyShop/Podfile [hunk 10]
```diff
@@ -5,7 +5,8 @@ target 'FancyShop' do
   platform :ios, '11.0'
 
   # Pods for MerchantApp
-  pod 'MCSCommerceWeb', :podspec => '../../MCSCommerceWeb.podspec'
+ # pod 'MCSCommerceWeb', :podspec => '../../MCSCommerceWeb.podspec'
```
Comment (cesar-oyarzun-m) [inline Example/FancyShop/Podfile]: Revert this to prev versions

File: Example/FancyShop/Pods/Target Support Files/Pods-FancyShop/Pods-FancyShop-frameworks-Debug-input-files.xcfilelist [hunk 11]
```diff
@@ -1,7 +1,7 @@
 ${PODS_ROOT}/Target Support Files/Pods-FancyShop/Pods-FancyShop-frameworks.sh
 ${BUILT_PRODUCTS_DIR}/Alamofire/Alamofire.framework
 ${BUILT_PRODUCTS_DIR}/CocoaLumberjack/CocoaLumberjack.framework
-${PODS_ROOT}/MCSCommerceWeb/MCSCommerceWeb.framework
```
Comment (cesar-oyarzun-m) [inline Example/FancyShop/Pods/Target Support Files/Pods-FancyShop/Pods-FancyShop-frameworks-Debug-input-files.xcfilelist:4]: Stop tracking pods files.

**Review by cesar-oyarzun-m — COMMENTED** [review]
I tested the pr and looks fine, just finish the pr comments and we can merge this PR

File: Example/FancyShop/FancyShop/Modules/OrderSummary/Application Logic/Interactor/OrderSummaryInteractor.swift (hunk 9 above)
Comment (BeansJeesh) [reply Example/FancyShop/FancyShop/Modules/OrderSummary/Application Logic/Interactor/OrderSummaryInteractor.swift]: comment added

File: Example/FancyShop/Podfile (hunk 10 above)
Comment (BeansJeesh) [reply Example/FancyShop/Podfile]: reverted

Comment (BeansJeesh) [reply Example/FancyShop/Podfile:8]: reverted

File: Example/FancyShop/Pods/Target Support Files/Pods-FancyShop/Pods-FancyShop-frameworks.sh (hunk 6 above)
Comment (BeansJeesh) [reply Example/FancyShop/Pods/Target Support Files/Pods-FancyShop/Pods-FancyShop-frameworks.sh]: reverted

Comment (BeansJeesh) [reply Example/FancyShop/Pods/Target Support Files/Pods-FancyShop/Pods-FancyShop-frameworks.sh]: don't think these are showing up in changed files now

File: Example/FancyShop/Podfile (hunk 10 above)
Comment (BeansJeesh) [reply Example/FancyShop/Podfile:8]: changed

File: Example/FancyShop/Pods/Target Support Files/Pods-FancyShop/Pods-FancyShop-frameworks-Debug-input-files.xcfilelist (hunk 11 above)
Comment (BeansJeesh) [reply Example/FancyShop/Pods/Target Support Files/Pods-FancyShop/Pods-FancyShop-frameworks-Debug-input-files.xcfilelist:4]: don't think we're tracking them anymore now

File: Example/FancyShop/FancyShop.xcodeproj/project.pbxproj (hunk 7 above)
Comment (BeansJeesh) [reply Example/FancyShop/FancyShop.xcodeproj/project.pbxproj:25]: I think this is mostly external components and configurations that are changed in here

File: Example/FancyShop/FancyShop/Modules/OrderSummary/Application Logic/Interactor/OrderSummaryInteractor.swift (hunk 9 above)
Comment (cesar-oyarzun-m) [reply Example/FancyShop/FancyShop/Modules/OrderSummary/Application Logic/Interactor/OrderSummaryInteractor.swift]: where is the comment I don't see it, Did you push your changes?

File: Example/SampleMerchantApp/SampleMerchantApp.xcodeproj/project.pbxproj [hunk 12]
```diff
@@ -874,4 +874,4 @@
 /* End XCConfigurationList section */
 	};
 	rootObject = B75202F5235775F500791723 /* Project object */;
-}
+}
```
Comment (cesar-oyarzun-m) [inline Example/SampleMerchantApp/SampleMerchantApp.xcodeproj/project.pbxproj]: Revert this file

File: Example/FancyShop/FancyShop.xcodeproj/project.pbxproj [hunk 13]
```diff
@@ -1,2400 +0,0 @@
-// !$*UTF8*$!
```
Comment (cesar-oyarzun-m) [inline Example/FancyShop/FancyShop.xcodeproj/project.pbxproj]: I think you need to put this file back, we need to track this file

File: MCSCommerceWebTests/MCSMockViewController.h [hunk 14]
```diff
@@ -0,0 +1,19 @@
...
+
+NS_ASSUME_NONNULL_BEGIN
+
+@interface MCSMockViewController : UIViewController
+
+@property (nonatomic) UIViewController *viewControllerPresented;
+
+@end
+
+NS_ASSUME_NONNULL_END
```
Comment (cesar-oyarzun-m) [inline MCSCommerceWebTests/MCSMockViewController.h]: remove this tags

**Comment by cesar-oyarzun-m** [comment]
This work will be taking care of in a different PR

**Pull request closed without merging**